azurelinuxagent/common/logcollector.py

```diff
@@ -19,59 +19,83 @@
 
 import glob
 from heapq import heappush, heappop
+import logging
 import os
 import subprocess
 import zipfile
 
+from azurelinuxagent.common.conf import get_lib_dir, get_ext_log_dir
 from azurelinuxagent.common.future import ustr
-from azurelinuxagent.common.utils.fileutil import mkdir, read_file, rm_files, append_file, append_items_to_file
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, write_file, rm_files
 
+_EXTENSION_LOG_DIR = get_ext_log_dir()
+_AGENT_LIB_DIR = get_lib_dir()
 
-_LOG_COLLECTOR_DIR = '/var/lib/waagent/logcollector'
-_TRUNCATED_FILES_DIR = '/var/truncated'
+_LOG_COLLECTOR_DIR = os.path.join(_AGENT_LIB_DIR, "logcollector")
+_TRUNCATED_FILES_DIR = "/var/truncated"
 
 _OUTPUT_RESULTS_FILE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "results.txt")
-COMPRESSED_ARCHIVE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "logs.zip")
+_COMPRESSED_ARCHIVE_PATH = os.path.join(_LOG_COLLECTOR_DIR, "logs.zip")
 
 _MUST_COLLECT_FILES = [
-    '/var/log/waagent.log',
-    '/var/lib/waagent/GoalState.*.xml',
-    '/var/lib/waagent/ExtensionsConfig.*.xml',
-    '/var/lib/waagent/waagent_status.json',
-    '/var/lib/waagent/history/*.zip',
-    '/var/log/azure/*/*',
-    '/var/log/azure/*/*/*',
-    '/var/lib/waagent/HostingEnvironmentConfig.xml',
-    '/var/log/waagent*',
+    "/var/log/waagent.log",
+    os.path.join(_AGENT_LIB_DIR, "GoalState.*.xml"),
+    os.path.join(_AGENT_LIB_DIR, "ExtensionsConfig.*.xml"),
+    os.path.join(_AGENT_LIB_DIR, "HostingEnvironmentConfig.*.xml"),
+    os.path.join(_AGENT_LIB_DIR, "SharedConfig.*.xml"),
+    os.path.join(_AGENT_LIB_DIR, "*manifest.xml"),
+    os.path.join(_AGENT_LIB_DIR, "waagent_status.json"),
+    os.path.join(_AGENT_LIB_DIR, "history", "*.zip"),
+    os.path.join(_EXTENSION_LOG_DIR, "*", "*"),
+    os.path.join(_EXTENSION_LOG_DIR, "*", "*", "*"),
+    "/var/log/waagent*",
 ]
 
 _FILE_SIZE_LIMIT = 30 * 1024 * 1024  # 30 MB
 _UNCOMPRESSED_ARCHIVE_SIZE_LIMIT = 150 * 1024 * 1024  # 150 MB
 
+_LOGGER = logging.getLogger(__name__)
+
 
 class LogCollector(object):
 
     _TRUNCATED_FILE_PREFIX = "truncated_"
 
     def __init__(self, manifest_file_path):
-        self.manifest_file_path = manifest_file_path
-        self.must_collect_files = self._expand_must_collect_files()
+        self._manifest_file_path = manifest_file_path
+        self._must_collect_files = self._expand_must_collect_files()
+        self._create_base_dirs()
+        self._set_logger()
+
+    @staticmethod
+    def _create_base_dirs():
+        mkdir(_TRUNCATED_FILES_DIR)
+        mkdir(_LOG_COLLECTOR_DIR)
 
     @staticmethod
-    def run_shell_command(command, stdout=subprocess.PIPE, output=False):
+    def _set_logger():
+        _f_handler = logging.FileHandler(_OUTPUT_RESULTS_FILE_PATH, encoding="utf-8")
+        _f_format = logging.Formatter(fmt='%(asctime)s %(levelname)s %(message)s',
+                                      datefmt='%Y-%m-%d %H:%M:%S')
```

Comment:
Could you make the dateformat consistent to our logs pls - 
`(u'%Y-%m-%dT%H:%M:%S.%fZ')`

Reply:
You should also set this to ensure that the time is always in UTC - 
`logging.Formatter.converter = time.gmtime`

Reply:
Thanks, good suggestions! Updated.